packages/astro/src/runtime/client/hmr.ts

```diff
@@ -1,38 +1,23 @@
 if (import.meta.hot) {
 	const parser = new DOMParser();
-	import.meta.hot.on('astro:reload', async ({ html }: { html: string }) => {
-		const { default: morphdom } = await import('morphdom');
+	import.meta.hot.on('astro:update', async ({ file }) => {
+		const { default: diff } = await import('micromorph');
+		// eslint-disable-next-line no-console
+		console.log(`[vite] hot updated: ${file}`)
+		import.meta.hot?.accept();
+		const html = await fetch(`${window.location}`).then(res => res.text())
 		const doc = parser.parseFromString(html, 'text/html');
+		
+		// Match incoming islands to current state
+		const roots = Array.from(doc.querySelectorAll('astro-root'));
```

Comment:
just a side note, but you don't have to convert this to an array, can use `for/of` on a NodeList.